python/cudf/cudf/core/dataframe.py

```diff
@@ -7478,12 +7478,32 @@ def cov(self, **kwargs):
         df.columns = self.columns
         return df
 
-    def corr(self):
+    def corr(self, method="pearson"):
         """Compute the correlation matrix of a DataFrame.
+
+        Parameters
+        ----------
+        method : str, Default 'pearson'
+            The correlation method to use, one of 'pearson' or 'spearman'.
+
+        Returns
+        -------
+        DataFrame
+            The requested correlation matrix.
         """
-        corr = cupy.corrcoef(self.values, rowvar=False)
-        df = DataFrame(cupy.asfortranarray(corr)).set_index(self.columns)
-        df.columns = self.columns
+        if method == "pearson":
+            corr = cupy.corrcoef(self.values, rowvar=False)
```

Comment:
Calling `values` directly may fail if the dataframe is nullable. The test case only captures the situation where the dataframe is non-nullable. Perhaps we should do a dropna here?

Reply:
Out of scope of this PR.